src/test/java/org/jabref/logic/importer/fetcher/CompositeSearchBasedFetcherTest.java

```diff
@@ -101,7 +112,15 @@ void performSearchOnNonEmptyQuery(Set<SearchBasedFetcher> fetchers) throws Excep
     static Stream<Arguments> performSearchParameters() {
         ImportFormatPreferences importFormatPreferences = mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS);
         ImporterPreferences importerPreferences = mock(ImporterPreferences.class);
+        BuildInfo buildInfo = new BuildInfo();
         when(importerPreferences.getApiKeys()).thenReturn(FXCollections.emptyObservableSet());
```

Comment:
Maybe remove this line to cause an error when an unknown key is retrieved? To catch errors early when a new fetcher is added.

Reply:
I'm not sure what you mean by this comment.  If I remove it the parameterized tests will start failing.

Should i create a whole new set of tests?

Reply:
No, keep as is.

Adress the other (minor) comments please - and then this should be good to go!